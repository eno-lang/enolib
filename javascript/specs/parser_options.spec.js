const eno = require('..');
const { HtmlReporter, TerminalReporter } = require('..');

describe('Parser options', () => {
  describe('sourceLabel', () => {
    it('is printed in text reports if provided', () => {
      expect(() => eno.parse('boom!', { source: '/some/dubious-file.eno' })).toThrowErrorMatchingSnapshot();
    });

    it('is printed in html reports if provided', () => {
      expect(() => eno.parse('boom!', { reporter: HtmlReporter, source: '/some/dubious-file.eno' })).toThrowErrorMatchingSnapshot();
    });

    it('is printed in terminal reports if provided', () => {
      expect(() => eno.parse('boom!', { reporter: TerminalReporter, source: '/some/dubious-file.eno' })).toThrowErrorMatchingSnapshot();
    });
  });
});
