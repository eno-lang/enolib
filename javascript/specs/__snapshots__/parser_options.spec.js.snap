// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser options sourceLabel is printed in html reports if provided 1`] = `
"Line 1 does not follow any specified pattern.

<div><div>/some/dubious-file.eno</div><pre class=\\"eno-report\\"><div class=\\"eno-report-line \\"><div class=\\"eno-report-gutter\\">      Line</div><div class=\\"eno-report-content\\">Content</div></div><div class=\\"eno-report-line eno-report-line-emphasized\\"><div class=\\"eno-report-gutter\\">         1</div><div class=\\"eno-report-content\\">boom!</div></div></pre></div>"
`;

exports[`Parser options sourceLabel is printed in terminal reports if provided 1`] = `
"Line 1 does not follow any specified pattern.

[30m[101m >      [0m [1m/some/dubious-file.eno[0m
[30m[101m >    1 [0m boom!"
`;

exports[`Parser options sourceLabel is printed in text reports if provided 1`] = `
"Line 1 does not follow any specified pattern.

-- /some/dubious-file.eno --

   Line | Content
 >    1 | boom!"
`;
